internal/factsengine/gatherers/sbd.go

```diff
@@ -0,0 +1,52 @@
+package gatherers
+
+import (
+	"github.com/joho/godotenv"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/cluster"
+)
+
+const (
+	SBDConfigGathererName = "sbd_config"
+	UndefinedSBDConfig    = "trento.checks.sbd.errors.undefined_configuration"
+)
+
+type SBDGatherer struct {
+	configFile string
```

Comment:
Maybe we could use afero to make this testable. It is used in other gatherers like this one. As reference: https://github.com/trento-project/agent/pull/78

Reply:
The absence of afero is just because of the usage of the https://github.com/joho/godotenv library, which under the hood uses simply `os.Whatever(...)`.
Since we are providing the configuration file path as a dependency, testing works pretty well, imho:
- at runtime the configuration file is the proper `cluster.SBDConfigPath` aka `cluster.SBDConfigPath`
- during tests the proper fake file is provided `../../../test/sbd_config`

The point is: if I can instruct the code to use a specific file (actual one or test one), I don't mind using the straight `os` file related functions, because I will just provide the correct file path in the correct scenario.
This thing, in this case at least, allows us to change the file we use, not the complete filesystem abstraction we are using, and I find it ok. Does this make sense?